### ISSUE TITLE
Auto Full Screen & Esc to Close Program (Not from GUI view from Fullscreen View)

### ISSUE THREAD

**Issue by Bonethugs1980** [nomacs/nomacs#550]
This is not a major problem, just a couple of minor 'quality of life' adjustments.  Two things of note:

1. When you press Esc button (from the settings 'Close on Esc'), it goes back to the GUI view - which is fine.  There is no option to close the program entirely from Esc...

2. There is no option to open and go straight into fullscreen mode either from clicking/opening a file directly or from opening the file from within the program 'Open'.  It always goes to GUI view first before pressing F11 or selecting 'View'>'Fullscreen' or double click on the image (dependent on setting).

*If possible for upcoming release(s) (not important right now), could an extra option be 'Esc closes program' (when selecting the this potential option it will untick 'Close on Esc' or 'Exit program in fullscreen with Esc'

*If possible have the program auto open to fullscreen from the start of opening a new applicable/supported image file.

If this is already in the roadmap or of things to do, please ignore and I can close if already planned.

**Comment by shankarsivarajan** [comment]
On version 3.17.2206, "Close on Esc" closes the program completely.

**Comment by Bonethugs1980** [comment]
Thanks for letting me know, Shall I just close this and mention the full screen from start issue.  Or leave this open and update the heading and comments as such?

**Comment by shankarsivarajan** [comment]
I don't know the etiquette around this either, but I'd leave it open and edit it. For future reference, I think it's good practice to make each bug/feature request a separate issue so they can be closed independently.

**Comment by Bonethugs1980** [comment]
Just installed the beta version you mentioned 3.17.2206.  This does 'Esc to Close' the same way.  Is there a misunderstanding?

So, similar to FastStone Image Viewer (not sure if I can mentioned here).  It does Esc to Close from fullscreen, where as nomacs does it back to the GUI view, which is one step extra instead of just closing it outright from fullscreen view.

**Comment by shankarsivarajan** [comment]
Ah, I see what you mean. I didn't understand the term "UI View"; I thought you meant that Esc closes the image but not the program, like Ctrl+W does. My bad.

**Comment by diemmarkus** [comment]
> 1. When you press Esc button (from the settings 'Close on Esc'), it goes back to the GUI view - which is fine.  There is no option to close the program entirely from Esc...

true, it only closes nomacs if not in fullscreen - that's actually a bug.

> 2\. There is no option to open and go straight into fullscreen mode either from clicking/opening a file directly or from opening the file from within the program 'Open'.  It always goes to GUI view first before pressing F11 or selecting 'View'>'Fullscreen' or double click on the image (dependent on setting).

There is a workaround for now: you can open a terminal and type:
```cmd
"C:\Program Files\nomacs\bin\nomacs.exe" --fullscreen
```

**Comment by yacoob** [comment]
Re `--fullscreen` - ideally, it'd be an option so that nomacs starts in fullscreen by default. It's good that a flag is here, but changing the commandline used for every associated filetype is not a viable approach here :)

**Comment by scrubbbbs** [comment]
There seems to be some attempt to restore fullscreen windows but it is bugged currently (master bf81943). Alt-F4 will exit the application from fullscreen with the default config file  on gnome, but weird things happen after that. I prepared a short clip to better explain [it.]

https://github.com/nomacs/nomacs/assets/89664476/b840a3a7-8ad8-48f6-8eca-d033c7be44e6

**Comment by scrubbbbs** [comment]
Closing as I believe we have a solution since #1136 . Nomacs now remembers the last state it was in at exit, so if it was fullscreen when closed before, it will re-open as such.

To make the fullscreen switch temporary (and all settings changes temporary), you can launch nomacs in private mode with "--private" or "File->Start Private Instance"
